zebra-utils/zcash-rpc-block-template-to-proposal

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Gets a block template from a Zcash node instance,
+# turns it into a block proposal using `block-template-to-proposal`,
+# and sends it to one or more Zcash node instances (which can include the same node).
+#
+# If there are multiple proposal ports, displays a diff of the responses.
+#
+# Uses `zcash-cli` with the RPC ports supplied on the command-line.
+
+function usage()
+{
+    echo "Usage:"
+    echo "$0 block-template-rpc-port proposal-rpc-port [extra-proposal-rpc-port...] -- [extra-block-template-rpc-json-fields] [extra-proposal-rpc-fields]"
+}
+
+# Override the commands used by this script using these environmental variables:
+ZCASH_CLI="${ZCASH_CLI:-zcash-cli}"
+DIFF="${DIFF:-diff --unified --color=always}"
+BLOCK_TEMPLATE_TO_PROPOSAL="${BLOCK_TEMPLATE_TO_PROPOSAL:-block-template-to-proposal}"
+
+# Process arguments
+
+if [ $# -lt 2 ]; then
+    usage
+    exit 1
+fi
+
+TEMPLATE_RPC_PORT=$1
+shift
+
+PROPOSAL_RPC_PORTS=""
+PROPOSAL_RPC_PORT_COUNT=0
+while [ -n "${1:-}" ] && [ "${1-}" != "--" ]; do
```

Comment:
```suggestion
if [ -n "${1:-}" ] && [ "${1-}" != "--" ]; then
    PROPOSAL_RPC_PORTS=$1
    PROPOSAL_RPC_PORT_COUNT=$((PROPOSAL_RPC_PORT_COUNT + 1))
    shift
fi

while [ -n "${1:-}" ] && [ "${1-}" != "--" ]; do
```

Reply:
I think the goal of this suggestion is to remove an extra space in `$PROPOSAL_RPC_PORTS`?

I don't really mind either way, but I prefer to avoid duplicating code.

Reply:
> I think the goal of this suggestion is to remove an extra space in $PROPOSAL_RPC_PORTS?

Yep, it would be nice to trim the whitespace but it's not too important.